pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -387,6 +394,14 @@ function validateParams(vmParams) {
         validationFailed['vmName'] = _("Name should not be empty");
     }
 
+    // If we select installation media from URL force the user to select
+    // OS, since virt-install will not detect the OS, in case we don't choose
+    // to start the guest immediately.
+    if (vmParams.vendor == NOT_SPECIFIED && vmParams.sourceType == URL_SOURCE && !vmParams.startVm)
+        validationFailed['vendor'] = _("OS vendor should be selected. " +
+        "Can't find the operating system you are looking for? " +
+        "Try selecting the next most recent version displayed instead of using the 'Unspecified' entry");
```

Comment:
grammar nitpick: should → must/need to, and perhaps let's avoid passive voice? This is also a tad confusing -- you can't select a vendor, but not an OS, as the OS will then always have some non-"unknown" default.

How about: "You need to select the most closely matching OS vendor and Operating System."?

Reply:
Sounds good!